docs/src/types.md

```diff
@@ -41,8 +41,10 @@ The `ValueSupport` sub-types defined in `Distributions.jl` are:
 
 **Type** | **Element type** | **Descriptions**
 --- | --- | ---
-`Discrete` | `Int` | Samples take discrete values
-`Continuous` | `Float64` | Samples take continuous real values
+`DiscreteSupport{T}` | `T` | Samples take any discrete values
+`Discrete = DiscreteSupport{Int}` | `Int` | Samples take `Int` values
+`ContinuousSupport{T <: Number}` | `T` | Samples take continuous values
+`Continuous = ContinuousSupport{Float64}` | `Float64` | Samples take continuous `Float64` values
```

Comment:
Should we use a `T <: Real` for Continuous?

Reply:
No, because we can have continuous distributions over things like complex numbers and Unitful numbers. It would be nice to have a continuous trait and have a constraint based on that, but this would be quite complex to implement, and can anyway be added in after this basic framework is up and running.

Reply:
"`T` is the `eltype` of the distribution", without specifying a subtype? This would allow other floating point, bignum, etc

Reply:
I'm not quite sure what you're suggesting here by the way? Is this good as is?

Reply:
We've agreed to leave this for now.